src/main/java/PeselValidator.java

```diff
@@ -0,0 +1,126 @@
+public class PeselValidator {
+
+
+    public boolean validate(String pesel){
+        return isLengthCorrect(pesel)
+                        && isFormatCorrect(pesel)
+                        && isMonthCorrect(pesel)
+                        && isDayCorrect(pesel)
+                        && isControlDigitCorrect(pesel);
+    }
+
+
+    public boolean validate(String pesel, String gender){
+        return validate(pesel) && whatGender(pesel).equals(gender);
+    }
+
+    private boolean isLengthCorrect(String pesel) {
+
+        if (pesel.length() == 11) return true;
+        else return false;
+    }
+
+
+    private boolean isMonthCorrect(String pesel) {
+
+        int monthNumber = Integer.valueOf(pesel.substring(2,4));
+
+        if (monthNumber > 0 && monthNumber < 13) return true;
+        if (monthNumber > 20 && monthNumber < 33) return true;
+        if (monthNumber > 40 && monthNumber < 53) return true;
+        if (monthNumber > 60 && monthNumber < 73) return true;
+        if (monthNumber > 80 && monthNumber < 93) return true;
+        else return false;
+    }
+
+
+    private String whatGender(String pesel) {
+
+        char genderDigit = pesel.charAt(9);
+
+        if (genderDigit == 0
+                || genderDigit == 2
+                || genderDigit == 4
+                || genderDigit == 6
+                || genderDigit == 8) return "female";
+        else return "male";
+    }
+
+
+    //checking if day number is correct assuming that February has always 29 days.
+    private boolean isDayCorrect(String pesel) {
+
+        int dayNumber = Integer.valueOf(pesel.substring(4,6));
+        int monthNumber = Integer.valueOf(pesel.substring(2,4));
+        int maxDaysInMonth = 0;
+
+        int[] monthsWith31Days = {
+                1,3,5,7,8,10,12,
+                21,23,25,28,30,32,
+                41,43,45,48,50,52,
+                61,63,65,68,70,72,
+                81,83,85,88,90,92};
+
+        int[] monthsWith30Days = {
+                4,6,9,11,
+                24,26,29,31,
+                44,46,49,51,
+                64,66,69,71,
+                84,86,89,91};
+
+        int[] monthsWith29Days = {2,22,42,62,82};
+
+        if (contains(monthsWith29Days,monthNumber)) maxDaysInMonth = 29;
+        else if (contains(monthsWith30Days,monthNumber)) maxDaysInMonth = 30;
```

Comment:
You may also consider Java 8 stream: Arrays.stream(monthsWith29Days).anyMatch(month -> month == monthNumber)